From e8c5303c0957e7e741840b3d945072876711b87c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:09 -0400
Subject: [kernel] ktime: add ms_to_ktime() and ktime_add_ms() helpers

Message-id: <1408953210-26343-2-git-send-email-jolsa@redhat.com>
Patchwork-id: 88725
O-Subject: [PATCH RHEL7.1 BZ1133083 001/322] ktime: add ms_to_ktime() and ktime_add_ms() helpers
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit d36f82b2435690d8742235d7bdc5bb5e878077e3
Author: Daniel Borkmann <dborkman@redhat.com>
Date: Tue Jun 25 18:17:26 2013 +0200

description
===========
Add two ktime helper functions that i) convert a given msec value to
a ktime structure and ii) that adds a msec value to a ktime structure.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/ktime.h b/include/linux/ktime.h
index bbca128..b4fa5e4 100644
--- a/include/linux/ktime.h
+++ b/include/linux/ktime.h
@@ -323,6 +323,11 @@ static inline ktime_t ktime_add_us(const ktime_t kt, const u64 usec)
  return ktime_add_ns(kt, usec * 1000);
 }
 
+static inline ktime_t ktime_add_ms(const ktime_t kt, const u64 msec)
+{
+ return ktime_add_ns(kt, msec * NSEC_PER_MSEC);
+}
+
 static inline ktime_t ktime_sub_us(const ktime_t kt, const u64 usec)
 {
  return ktime_sub_ns(kt, usec * 1000);
@@ -366,7 +371,15 @@ extern void ktime_get_ts(struct timespec *ts);
 static inline ktime_t ns_to_ktime(u64 ns)
 {
  static const ktime_t ktime_zero = { .tv64 = 0 };
+
  return ktime_add_ns(ktime_zero, ns);
 }
 
+static inline ktime_t ms_to_ktime(u64 ms)
+{
+ static const ktime_t ktime_zero = { .tv64 = 0 };
+
+ return ktime_add_ms(ktime_zero, ms);
+}
+
 #endif
-- 
1.7.1